protocol/linkpreview/linkpreview.go

```diff
@@ -168,19 +172,38 @@ func (u OEmbedUnfurler) unfurl() (common.LinkPreview, error) {
 	}
 
 	var oembedResponse OEmbedResponse
+	var giphyOembed giphyOembedResponse
 	if err != nil {
 		return preview, err
 	}
-	err = json.Unmarshal(oembedBytes, &oembedResponse)
+
+	type responseOembed struct {
+		title string
+		url   string
+	}
+
+	var response responseOembed
+	if strings.Contains(u.url.String(), "giphy") {
```

Comment:
@ibrkhalil, this kind of code defeats the purpose of the generic oembed interface. We already detected in a reliable way if it's a giphy URL in `newUnfurler` via a normalized hostname. Besides that, this check here is not reliable. After an unfurler is created by `newUnfurler` the code should be agnostic of any given host.

If the oembed response can vary between providers (which is definitely possible) then I think we should simply evolve the abstractions & interfaces to support this need.

Reply:
Please take another look Icaro when you have the time